added compiler headers

rmujica opened the pull request:
added compiler headers depending on which compiler is being used. this fixes a problem when trying to complile using pgsql

rmujica committed:
pgsql fix

ivucica commented on src/definitions.h:
```diff
@@ -58,6 +58,19 @@
+  #else
+    #include "compiler/gcc.h"
+    #ifdef __MINGW32__
+      #include "compiler/mingw32.h"
```
There is no such header.

ivucica reviewed (changes requested):
When creating pull requests like these, please ensure your changes are selfcontained and justified.

It's unclear to me why b894002 is necessary; please clarify which problem you're trying to clean up. If it is unnecessary, please use `git rebase -i` to eliminate it from the patchset.

I am also concerned that you're seemingly introducing the only uses of these compiler-specific headers present in the source code; grepping for `gcc.h`, `llvm.h` and `msvc.h` reveals no results. Can we instead get rid of these headers and resolve your issue some other way?

Other two commits seem sensible, albeit the commit descriptions could have been more detailed.

rmujica committed:
fix(pgsql): using atoll instead of macro ATOI64

rmujica commented:
Hi @ivucica , sorry for the delay. I have removed the commit about the compiler headers, I was using that because the PostgreSQL database driver used a macro (ATOI64) that I thought it was compiler-dependant. I have now changed the macro into `atoll` like the MySQL driver.

Thanks for your feedback! I hope to continue improving my pull requests as I develop my server further.

ivucica commented:
Thanks for your contributions!

Unfortunately, that commit is still present in the history you're pushing into this pull request, and neither have you removed your changes in 0cc882e. See all the changes you're trying to push here: https://github.com/opentibia/server/pull/59/files

Additionally, I'll still ask you to rebase your changes.

rmujica committed:
fix(pgsql): removed executeQuery from pgsql driver header

rmujica commented:
@ivucica sorry, it was my first time rebasing, I dropped the wrong commit. I have fixed that and commited the code I dropped.

ivucica commented:
I've slightly misinterpreted the Github web UI, and this is and was correctly rebased.

You've correctly removed the unnecessary changes, thanks.

I would have liked to see the commit log messages more descriptive, but I'll accept this. Thank you!

rmujica commented:
I will try to make commit messages more descriptive, thanks for your feedback!
